### DISCUSSION TITLE
The maxDuration parameter doesn't work for me

### DISCUSSION THREAD

**Discussion by v-ahuja** [vercel/next.js#66269] [Help]
### Summary

The `maxDuration` setting in the api route config, as per the documentation, isn't being honored on dev or on Vercel. We're on the pro plan.

We're using pages router and next version - 14.2.3

I have the following in my api route code:

```
export const config = {
  maxDuration: 2,
  api: {
    bodyParser: false, // Disallow body parsing, consume as stream
  },
};
```

There are 2 things I would expect to happen:

1. on dev, i.e. via `next dev`, i would expect to see that route timeout after 2s. It does not.

2. When deploying to vercel, i would expect the `maxDuration` correctly reflected in the `Function Details`. While deploying, I have set the `maxDuration` to be 35s

<img width="1223" alt="image" src="https://github.com/vercel/next.js/assets/5714957/9ea3a0af-b2e2-49fc-88a4-caff605fa559">

### Additional information

```js
`next info`


Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:22 PDT 2022; root:xnu-8020.121.3~4/RELEASE_X86_64
  Available memory (MB): 16384
  Available CPU cores: 12
Binaries:
  Node: 21.7.3
  npm: 10.5.0
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: 13.3.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.2
Next.js Config:
  output: N/A
```

**Comment by icyJoseph** [comment]
Hi,

Is this within a Route handler (app router)? The config object export, only worked for API Routes (pages router, IIRC)

**Reply by v-ahuja** [reply]
hey, it's pages router. i'll update the discussion. tks for pointing that out.
